Tests/src/artofillusion/SceneTest.java

```diff
@@ -10,32 +10,768 @@
+        public void itemRemoved(int index, Object obj)
+        {
+          listenerFireCount++;
+        }
+
+        @Override
+        public void itemChanged(int index, Object obj)
+        {
+        }
+      });
+      
+      scene.removeMaterial(0);
+
+      assertEquals(0, scene.getNumMaterials());
+      assertEquals(1, listenerFireCount);
+      
+      assertNull(target.getObject().getMaterial());
+      assertNull(target.getObject().getMaterialMapping());
+      
+    }
+    
+  /**
+   * Test to check itemChanged event is fired on material change
+   */
+  @Test
+    public void testChangeMaterialEventFired()
+    {
+
+      Material mat = new UniformMaterial();
+      scene.addMaterial(mat);
+      listenerFireCount = 0;
+
+      
+      scene.addMaterialListener( new ListChangeListener()
+      {
+        @Override
+        public void itemAdded(int index, Object obj)
+        {
+        }
+
+        @Override
+        public void itemRemoved(int index, Object obj)
+        {
+        }
+
+        @Override
+        public void itemChanged(int index, Object obj)
+        {
+          listenerFireCount++;
+        }        
+      });
+      
+      scene.changeMaterial(0);
+      assertEquals(1, listenerFireCount);
+      
+    }
+
+  /**
+   * Test to get added material by name
+   */
+  @Test
+    public void testGetMaterialByName()
+    {
+    
+      Material mat = new UniformMaterial();
+      mat.setName("Test");
+      scene.addMaterial(mat);
+      mat = null;
+      mat = scene.getMaterial("Test");
+      assertNotNull(mat);
+    }
+    
+  /**
+   * Test not to get material by wrong name
+   */
+  @Test
+    public void testGetUnknownMaterialByName()
     {
+      Material mat = scene.getMaterial("Missing");
+      assertNull(mat);
     }
+    
+  /**
+   * Test not to get material by null name
+   */
+  @Test
+    public void testGetMaterialByNullName()
+    {
+      Material mat = scene.getMaterial(null);
+      assertNull(mat);
+    }
+    
+  /**
+   * Add texture to scene
+   * Check that texture listener event is triggered
+   */
+  @Test
+    public void testAddTextureEventFired() {
+
+      Texture tex = new UniformTexture();
+      listenerFireCount = 0;
+      
+      scene.addTextureListener( new ListChangeListener() {
+        @Override
+        public void itemAdded(int index, Object tex)
+        {
+          listenerFireCount++;
+        }
+
+        @Override
+        public void itemRemoved(int index, Object tex)
+        {
+        }
 
+        @Override
+        public void itemChanged(int index, Object tex)
+        {
+        }
+      });
+      
+      scene.addTexture(tex);
+      assertEquals(1, listenerFireCount);
+      assertEquals(2, scene.getNumTextures());
+    }
+    
+  /**
+   * Add texture to scene at given position.
+   * Check that texture listener event is triggered
+   * Check that texture is inserted at expected position
+   */
+  @Test
+    public void testAddTextureAtGivenPos()
```

Comment:
And this one as well. It seems to have the same logic as the above, and shows the same failure.